pkg/operation/botanist/controlplane/konnectivity/konnectivity_server.go

```diff
@@ -0,0 +1,560 @@
+						MaxSurge:       &intstr.IntOrString{Type: intstr.Int, IntVal: 1},
+					},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"checksum/secret-" + k.secrets.Kubeconfig.Name: k.secrets.Kubeconfig.Checksum,
+							"checksum/secret-" + k.secrets.Server.Name:     k.secrets.Server.Checksum,
+							"checksum/secret-" + k.secrets.ClientCA.Name:   k.secrets.ClientCA.Checksum,
+						},
+						Labels: utils.MergeStringMaps(getLabels(), map[string]string{
+							v1beta1constants.LabelNetworkPolicyToDNS:              v1beta1constants.LabelNetworkPolicyAllowed,
+							v1beta1constants.LabelNetworkPolicyFromShootAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+							v1beta1constants.LabelNetworkPolicyToShootAPIServer:   v1beta1constants.LabelNetworkPolicyAllowed,
+							v1beta1constants.LabelNetworkPolicyToSeedAPIServer:    v1beta1constants.LabelNetworkPolicyAllowed,
+						}),
+					},
+					Spec: corev1.PodSpec{
+						Affinity: &corev1.Affinity{
+							PodAntiAffinity: &corev1.PodAntiAffinity{
+								PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{{
+									Weight: 100,
+									PodAffinityTerm: corev1.PodAffinityTerm{
+										TopologyKey:   corev1.LabelHostname,
+										LabelSelector: &metav1.LabelSelector{MatchLabels: getLabels()},
+									},
+								}},
+							},
+						},
+						ServiceAccountName: name,
+						Containers: []corev1.Container{{
+							Name:            name,
+							Image:           k.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Args: []string{
+								fmt.Sprintf("--namespace=%s", k.Namespace),
+								fmt.Sprintf("--deployment-name=%s", name),
+								"--jitter=10s",
+								"--jitter-factor=5",
+								"--v=2",
+								"--",
+								"/proxy-server",
+								"--logtostderr=true",
+								fmt.Sprintf("--cluster-cert=%s/konnectivity-server.crt", serverMountPath),
+								fmt.Sprintf("--cluster-key=%s/konnectivity-server.key", serverMountPath),
+								fmt.Sprintf("--server-cert=%s/konnectivity-server.crt", serverMountPath),
+								fmt.Sprintf("--server-key=%s/konnectivity-server.key", serverMountPath),
+								fmt.Sprintf("--server-ca-cert=%s/ca.crt", clientCAMountPath),
+								fmt.Sprintf("--agent-namespace=%s", metav1.NamespaceSystem),
+								fmt.Sprintf("--agent-service-account=%s", AgentName),
+								fmt.Sprintf("--kubeconfig=%s/kubeconfig", kubeconfigMountPath),
+								fmt.Sprintf("--authentication-audience=%s", ServerAudience),
+								"--keepalive-time=1h",
+								"--log-file-max-size=0",
+								"--delete-existing-uds-file=true",
+								"--mode=http-connect",
+								fmt.Sprintf("--server-port=%d", ServerHTTPSPort),
+								fmt.Sprintf("--agent-port=%d", ServerAgentPort),
+								fmt.Sprintf("--admin-port=%d", adminPort),
+								fmt.Sprintf("--health-port=%d", healthPort),
+								"--v=2",
```

Comment:
```suggestion
								"--v=2",
								"--server-count",
```

Missing the last argument for replica-reloader. Will default to 1 instead of the correct number of replicas of the connectivity server deployment.